src/pgduckdb_hooks.cpp

```diff
@@ -169,10 +182,16 @@ IsAllowedStatement(Query *query, bool throw_error) {
 	/* We don't support modifying statements on Postgres tables yet */
 	if (query->commandType != CMD_SELECT) {
 		RangeTblEntry *resultRte = list_nth_node(RangeTblEntry, query->rtable, query->resultRelation - 1);
-		if (!::IsDuckdbTable(resultRte->relid)) {
+		if (::IsHeapTable(resultRte->relid) && query->commandType == CMD_INSERT) {
```

Comment:
Why only heap tables? I'd expect this should logic would work fine for most table access method, because they implement the table access method interface.

Reply:
YES, you are right! This is becoming much more exciting if we get the whole thing functional.